compiler/src/iree/compiler/Codegen/Interfaces/BufferizationInterfaces.cpp

```diff
@@ -70,22 +70,23 @@ getMemrefTypeForTensor(IREE::Flow::DispatchTensorType tensorType,
                          tensorType.getBoundElementType(), layout, memorySpace);
 }
 
-/// Find the memref version of the given InterfaceBindingSubspanOp. If no such
-/// op exists in the same block (before the given op), create a new op.
-// TODO(#12933): Because of regressions in CUDA backend, there is an
-// option to keep a legacy mode of not representing the offset in the
-// type. Remove once the bug is fixed.
-static Value
-findOrCreateSubspanBuffer(RewriterBase &rewriter,
-                          IREE::HAL::InterfaceBindingSubspanOp subspanOp) {
+/// Compute the future buffer tyoe if a tensor subspan op.
+// TODO: This function should not modify any IR.
+static MemRefType
+computeSubspanBufferType(IREE::HAL::InterfaceBindingSubspanOp subspanOp) {
   // Ensure that this a tensor subspan op.
   auto shapedType = llvm::dyn_cast<IREE::Flow::DispatchTensorType>(
       subspanOp.getResult().getType());
   assert(shapedType && shapedType.hasRank());
 
+  IRRewriter rewriter(subspanOp.getContext());
+  rewriter.setInsertionPoint(subspanOp);
+
   Value byteOffset = subspanOp.getByteOffset();
   MemRefLayoutAttrInterface layoutAttr = {};
   if (byteOffset && !matchPattern(byteOffset, m_Zero())) {
+    // TODO: Find a way to call convertByteOffsetToElementOffset without a
```

Comment:
Do you see a way to address this?

Reply:
yeah you should be able to just get the type of the memref, leaving offset as dynamic. It should be possible to write a version of that method which just gets the type of the memref....